### PR TITLE
Rely primarily on Promises

### PR DESCRIPTION
**Pull request by jdfreder**
Problem: Code was overly complicated because of Observables

Solution: Simplify by using Promises first, then Observables

### CODE REVIEW

**Comment by rgbkrk** [comment]
As spoken in chat, @jdfreder doesn't want to merge this yet. Since all those relying on this so far are using the promises based approach, we're going to just use promises and get rid of the RxJS 4 dependency.

**Comment by rgbkrk** [comment]
Greatly simplified.

File: index.js [hunk 1]
```diff
@@ -1,86 +1,64 @@
...
-                    .filter(x => !x.err)
-                    .publish()
-                    .refCount()
+/**
+ * Gets a list of kernelInfo objects for a given directory of kernels
+ * @param  {string}   directory path to a directory full of kernels
+return return     files.map(fileName => ({
+      name: fileName,
+      resourceDir: path.join(directory, fileName)
+    }))
```
Comment (jdfreder) [inline index.js:44]: EW @rgbkrk atom bug

Comment (rgbkrk) [inline index.js:44]: Yup.

File: index.js [hunk 2]
```diff
@@ -1,86 +1,64 @@
...
+ * @return {Promise<object[]>} Promise for an array of kernelResources objects
+ */
+function kernelSpecs() {
+  return jp.dataDirs({ withSysPrefix: true }).then(dirs => {
+    return Promise.all(dirs
+      .map(dir => getKernelInfos(dir)) // get kernel infos for each directory
+      .reduce((a, b) => a.concat(b)) // flatten the results into one array
+      .map(a => a.catch())) // ignore kernelInfo related errors
+      .map(a => getKernelResources(a)) // convert kernelInfo -> kernelResources
+      .map(a => a.catch()) // ignore kernelResources related errors
```
Comment (rgbkrk) [inline index.js:59]: This is why I wish for streams in general - stream of errors and a stream of valid data.

**Comment by rgbkrk** [comment]
We'll be calling this one a major release since we updated the interface. In the future we should retain backwards compat. Since this is in draft stage at the moment and only a couple projects using it, we're fit to make this a clean, trim interface.

**Comment by rgbkrk** [comment]
Awesome note @jdfreder - using native promises here sped the code up quite a bit: ~0.225s for `v0.2.0` (relied on Rx) and ~0.119s for `master`.

**Pull request merged**